BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/location/LocationUtil.java

```diff
@@ -0,0 +1,62 @@
+package com.nexenio.bleindoorpositioning.location;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by leon on 29.03.18.
+ */
+
+public class LocationUtil {
+
+    public static Location meanLocationFromLast(List<Location> locationList, long amount, TimeUnit timeUnit) {
+        List<Location> matchingLocations = getLocationsFromLast(locationList, amount, timeUnit);
+        return meanLocation(matchingLocations);
+    }
+
+    public static Location meanLocation(List<Location> locationList) {
+        if (locationList.size() > 1) {
+            Location meanLocation = new Location();
+            double latitudeSum = 0;
+            double longitudeSum = 0;
+            for (int i = 0; i < locationList.size(); i++) {
+                latitudeSum += locationList.get(i).getLatitude();
+                longitudeSum += locationList.get(i).getLongitude();
+            }
+            meanLocation.setLatitude(latitudeSum / locationList.size());
+            meanLocation.setLongitude(longitudeSum / locationList.size());
+            return meanLocation;
+        } else if (locationList.size() == 1) {
+            return locationList.get(0);
+        } else {
+            return null;
+        }
+    }
+
+    public static List<Location> getLocationsBetween(List<Location> locationList, long startTimestamp, long stopTimestamp) {
+        List<Location> matchingLocations = new ArrayList<>();
+        for (int i = 0; i < locationList.size(); i++) {
```

Comment:
Use `for (Location location : locationList) {` instead of using multiple lookups.